src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Machine.class) {
+            return List.of();
+        }
+
+        try {
+            try {
+                return List.of(type.getDeclaredConstructor().newInstance());
+            } catch (InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException(e);
+            }
+        } catch (InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
```

Comment:
The `fill` method uses `Machine` as the type for `value`, which limits its flexibility. If the class is parameterized with `T`, consider changing the parameter type to `T`.